### PR TITLE
#39 basic exec module

### LINKED ISSUE #39: Init exec module

### PR DESCRIPTION
**Pull request by alex216**
fix #39

## Summary by CodeRabbit

- **New Features**
	- Introduced process management capabilities with the ability to execute shell commands asynchronously.
	- Added a new interface for spawning processes, enhancing the application's functionality.
	- Implemented readline functionality for improved command line input and history management.

- **Bug Fixes**
	- Removed outdated command execution methods to streamline process management.

- **Tests**
	- Added unit tests for the new process spawning functionality to ensure expected behavior.

### CODE REVIEW

**Review by rask24 — COMMENTED** [review]
@alex216 check my comments.

File: src/main.c [hunk 1]
```diff
@@ -3,63 +3,21 @@
...
-#include <errno.h>
-#include <signal.h>
-#include <stdbool.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
-#include <sys/wait.h>
-#include <termios.h>
-#include <unistd.h>
+#include <exec.h>
```
[the remark this answers is not in the document]
Comment (alex216) [reply src/main.c:13]: d87959d Fix header error

[the remark this answers is not in the document]
Comment (rask24) [reply src/main.c:13]: Ref: https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html

File: src/exec/exec.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+/*   exec.c                                             :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: yliu <yliu@student.42.jp>                  +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/09/06 12:54:34 by yliu              #+#    #+#             */
+/*   Updated: 2024/09/06 14:32:48 by yliu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <exec.h>
```
[the remark this answers is not in the document]
Comment (alex216) [reply src/exec/exec.c]: ed240d6 Add readline.h to refactor header dependency

File: include/exec.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# define EXEC_H
+
+# include "ui.h"
+# include "utils.h"
+# include <errno.h>
+# include <stdio.h>
+# include <stdlib.h>
+# include <string.h>
+# include <sys/wait.h>
+# include <unistd.h>
```
[the remark this answers is not in the document]
Comment (alex216) [reply include/exec.h]: ed240d6 Add readline.h to refactor header dependency

**Review by alex216 — COMMENTED** [review]
@rask24 may check the commit.

**Review by rask24 — APPROVED** [review]
@alex216 LGTM 🚀

**Pull request merged**